python/pyarrow/_compute.pyx

```diff
@@ -1736,6 +1736,28 @@ class PartitionNthOptions(_PartitionNthOptions):
         self._set_options(pivot, null_placement)
 
 
+cdef class _CumulativeGenericOptions(FunctionOptions):
+    def _set_options(self, start, skip_nulls):
+        self.wrapped.reset(new CCumulativeGenericOptions(
+            pyarrow_unwrap_scalar(start), skip_nulls))
+
+
+class CumulativeGenericOptions(_CumulativeGenericOptions):
+    """
+    Options for `cumulative_sum` function.
+
+    Parameters
+    ----------
+    start : Scalar
```

Comment:
Add a Python test because input is Scalar and need to ensure it is handled correctly.